middlewares/osio/auth.go

```diff
@@ -178,12 +179,26 @@ func (a *OSIOAuth) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.
 			}
 
 			if err != nil {
-				log.Errorf("Cache resole failed, %v", err)
+				log.Errorf("Cache resolve failed, %v", err)
 				rw.WriteHeader(http.StatusUnauthorized)
 				return
 			}
-			r.Header.Set("Target", cached.Location)
-			r.Header.Set("Authorization", "Bearer "+cached.Token)
+
+			stripServicePathPrefix(r, service)
+			targetURL := getServiceURL(cached.Namespace, service)
+			targetURL = normalizeURL(targetURL)
+
+			if isRedirectService(service) {
+				redirectURL := targetURL + r.URL.Path
+				if service == logs && r.URL.RawQuery != "" {
+					redirectURL = strings.Join([]string{redirectURL, "?", r.URL.RawQuery}, "")
+				}
+				http.Redirect(rw, r, redirectURL, http.StatusTemporaryRedirect)
+				return
```

Comment:
for call to console, logs service will be re-directed (no further processing, next handler will NOT be called).